pandas/core/indexes/base.py

```diff
@@ -631,6 +631,9 @@ def astype(self, dtype, copy=True):
         Index
             Index with values cast to specified dtype.
         """
+        if dtype is not None:
```

Comment:
is possible we need to optimize this at some point (pandas_dtype) to fast path things